service/scaladsl/client/src/main/scala/com/lightbend/lagom/scaladsl/client/ServiceClient.scala

```diff
@@ -140,28 +146,165 @@ trait LagomServiceClientComponents extends TopicFactoryProvider { self: LagomCon
+ * this standalone factory will create and manage an [[akka.actor.ActorSystem]] and Akka Streams [[akka.stream.Materializer]].
+ *
+ * It is important to invoke [[StandaloneLagomClientFactory#stop()]] when the application is no longer needed,
+ * as this will trigger the shutdown of the underlying [[akka.actor.ActorSystem]] and Akka Streams [[akka.stream.Materializer]]
+ * releasing all thread and connection pools in use by the clients.
+ *
+ * There is one more component that you’ll need to provide when creating a client application, that is a service locator.
+ * It is up to you what service locator you use, it could be a third party service locator, or a service locator created
+ * from static configuration.
+ *
+ * Lagom provides a number of built-in service locators, including a [[StaticServiceLocator]], a [[RoundRobinServiceLocator]]
+ * and a [[ConfigurationServiceLocator]]. The easiest way to use these is to mix in their respective Components traits.
+ *
+ * For example, here’s a client application built using the static service locator, which uses a static URI:
+ * {{{
+ * import java.net.URI
+ * import com.lightbend.lagom.scaladsl.client._
+ * import play.api.libs.ws.ahc.AhcWSComponents
+ *
+ * val clientApplication = new StandaloneLagomClientFactory("my-client")
+ *   with StaticServiceLocatorComponents
+ *   with AhcWSComponents {
+ *
+ *   override def staticServiceUri = URI.create("http://localhost:8080")
+ * }
+ * }}}
+ *
+ *
+ * @param clientName The name of the service that is consuming the Lagom service. This will impact how calls made through clients
+ *                   generated by this factory will identify themselves.
+ * @param classLoader A classloader, it will be used to create the service proxy and needs to have the API for the client in it.
+ */
+abstract class StandaloneLagomClientFactory(
+  clientName:  String,
+  classLoader: ClassLoader = classOf[StandaloneLagomClientFactory].getClassLoader
+) extends LagomClientFactory(clientName, classLoader) {
+
+  lazy val actorSystem: ActorSystem = new ActorSystemProvider(environment, configuration, applicationLifecycle).get
+  override lazy val materializer: Materializer = ActorMaterializer.create(actorSystem)
+
+  /**
+   * Stop this [[LagomClientFactory]] by shutting down the internal [[akka.actor.ActorSystem]], Akka Streams [[akka.stream.Materializer]]
+   * and internal resources.
+   */
+  override def stop(): Unit = {
+
+    // we need to use the stop method from ApplicationLifecycle because the
+    // WebSocket client register a stop hook on it
+    // ideally, we should have used a CoordinatedShutdown phase for it,
+    // but we can't know if the ActorSystem is going to be shutdown together with this Factory
+    implicit val ex = executionContext
+    val stopped = Future.sequence(
+      releaseInternalResources() +: actorSystem.terminate() +: Nil
+    ).map(_ => ())
```

Comment:
@ignasi35, I changed it to `Future.sequence` so they are not depending on each other. 

I'm also explicitly mapping to `Unit`. It's not good to use discarded values. We should enable the compiler flag to avoid it. Nasty bugs live behind discarded values. 

We must add the same to `master`. I have it ready and will send a PR as soon as this one gets approved.

Reply:
It'd be great if we could hold the execution of `actorSystem.terminate` until `releaseInternalResources` completed but I guess we'll be fine in most situation.

Reply:
We can do it, but then we need to recover if `releaseInternalResources` and then`flatMap` 

```scala
releaseInternalResources()
  .recover { ... }
  .flatMap { _ =>
    actorSystem.terminate()
  }
```

Reply:
Ah, I just realised `transformWith` is only available in scala 2.12 so we still can't use it. Then, yes: `recover` + `flatMap` is the only option we can use.